### PR TITLE
ipmi: Set default cipher suite to 17

### PR DESCRIPTION
**Pull request by Kwongyhue**
## What does this PR implement/change/remove?
When constructing commands executed by ipmitool, the default unspecified cipher suite is cipher suite 3. This PR will default to using the strong cipher suite 17 and defer to the universally supported IPMI 2.0 required cipher suite 3

### The BMC firmware and/or BIOS versions that this change applies to (if applicable)
OpenBMC only supports cipher suite 17 by default while more widely available BMC firmwares support the IPMI 2.0 required cipher suite 3

### What version of tooling - vendor specific or opensource does this change depend on (if applicable)
Tested on ipmitool version: 1.8.18-10

## Description for changelog/release notes
When constructing ipmitool command, will attempt to use cipher suite 17 before using cipher suite 3.
```
```

### CODE REVIEW

File: internal/ipmi/ipmi.go [hunk 1]
```diff
@@ -46,14 +49,21 @@ func (i *Ipmi) run(ctx context.Context, command []string) (output string, err er
...
-		return string(out), ctx.Err()
-	}
-	return string(out), errors.Wrap(err, strings.TrimSpace(string(out)))
+    for _, cipherString := range ipmiCiphers {
+        ipmiCmd := append(ipmiArgs, cipherString)
+        ipmiCmd = append(ipmiCmd, command...)
+        cmd := exec.CommandContext(ctx, i.ipmitool, ipmiCmd...)
+        cmd.Env = []string{fmt.Sprintf("IPMITOOL_PASSWORD=%s", i.Password)}
+        out, err = cmd.CombinedOutput()
+        if err == nil {
```
Comment (joelrebel) [inline internal/ipmi/ipmi.go:58]: The err check should include a check for context errors in the loop, 
```go
if err == nil || ctx.Err() != nil {
  break
}
```

File: internal/ipmi/ipmi.go [hunk 2]
```diff
@@ -46,14 +49,21 @@ func (i *Ipmi) run(ctx context.Context, command []string) (output string, err er
...
+            break
+        }
+    }
+
+    if ctx.Err() == context.DeadlineExceeded {
+        return string(out), ctx.Err()
+    }
+    return string(out), errors.Wrap(err, strings.TrimSpace(string(out)))
```
Comment (joelrebel) [inline internal/ipmi/ipmi.go:66]: nit: add whitespace before return

**Review by joelrebel — CHANGES_REQUESTED** [review]
Thanks for this cipher fix! it looks fine to me overall. I'll test this out on a few machines and report back.

File: internal/ipmi/ipmi.go (hunk 1 above)
Comment (Kwongyhue) [reply internal/ipmi/ipmi.go]: Amended

File: internal/ipmi/ipmi.go (hunk 2 above)
Comment (Kwongyhue) [reply internal/ipmi/ipmi.go:67]: Done

**Comment by joelrebel** [comment]
Tested this out on a bunch of hardware and it seems to work great, thanks!

```
❯ ./bmclib-tester run --tests samples/tests-ipmi.yaml --hardware hardware.yaml
2023/05/03 10:48:19 waiting for tests to complete...
[
 {
  "Vendor": "Supermicro",
  "Model": "SYS-510T-MR",
  "Name": "c",
  "BMCIP": "192.168.157.35",
  "Results": [
   {
    "Feature": "powerstate",
    "Protocol": "ipmi",
    "Output": "Chassis Power is on\n",
    "Error": "",
    "Succeeded": true,
    "Runtime": "1ns"
   }
  ]
 },
 {
  "Vendor": "Supermicro",
  "Model": "6029P-E1CR12L",
  "Name": "d",
  "BMCIP": "192.168.156.30",
  "Results": [
   {
    "Feature": "powerstate",
    "Protocol": "ipmi",
    "Output": "Chassis Power is on\n",
    "Error": "",
    "Succeeded": true,
    "Runtime": "1ns"
   }
  ]
 },
 {
  "Vendor": "Dell",
  "Model": "R6515",
  "Name": "e",
  "BMCIP": "192.168.160.11",
  "Results": [
   {
    "Feature": "powerstate",
    "Protocol": "ipmi",
    "Output": "Chassis Power is on\n",
    "Error": "",
    "Succeeded": true,
    "Runtime": "1ns"
   }
  ]
 },
 {
  "Vendor": "ASRockRack",
  "Model": "EPYC 7402P",
  "Name": "a",
  "BMCIP": "192.168.151.151",
  "Results": [
   {
    "Feature": "powerstate",
    "Protocol": "ipmi",
    "Output": "Chassis Power is on\n",
    "Error": "",
    "Succeeded": true,
    "Runtime": "1ns"
   }
  ]
 },
 {
  "Vendor": "Dell",
  "Model": "R750",
  "Name": "b",
  "BMCIP": "192.168.153.12",
  "Results": [
   {
    "Feature": "powerstate",
    "Protocol": "ipmi",
    "Output": "Chassis Power is off\n",
    "Error": "",
    "Succeeded": true,
    "Runtime": "1ns"
   }
  ]
 }
]

```

**Review by joelrebel — APPROVED** [review]

**Pull request merged**